kafka/producer_test.go

```diff
@@ -40,9 +45,16 @@ func TestProducerAPIs(t *testing.T) {
 	topic1 := "gotest"
 	topic2 := "gotest2"
 
+	envelope := Envelope{M: "Own drChan"}
+	data, err := json.Marshal(envelope)
+
+	if err != nil {
+		t.Error(err)
+	}
+
 	// Produce with function, DR on passed drChan
 	err = p.Produce(&Message{TopicPartition: TopicPartition{Topic: &topic1, Partition: 0},
-		Value: []byte("Own drChan"), Key: []byte("This is my key")},
+		Value: data, Key: []byte("This is my key")},
```

Comment:
Please add a new test action instead of changing existing since we want to be sure to maintain previous functionality too.